### PR TITLE
Move HPXMLtoOpenStudio to top level

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Companion: https://github.com/NREL/resstock/pull/971

Also consolidates all existing debug arguments (for example, those for `build_existing_model` and `server_directory_cleanup`) into a single `debug` argument stored at the top level of the workflow generator. It optionally enables debug mode across multiple workflow measures. See the companion PR for more information.

## Checklist

Not all may apply

- [ ] Code changes (must work)
- [ ] Tests exercising your feature/bug fix (check coverage report on Checks -> BuildStockBatch Tests -> Artifacts)
- [ ] Coverage has increased or at least not decreased. Update `minimum_coverage` in `.github/workflows/ci.yml` as necessary.
- [ ] All other unit tests passing
- [ ] Update validation for project config yaml file changes
- [ ] Update existing documentation
- [ ] Run a small batch run to make sure it all works (local is fine, unless an Eagle specific feature)
- [ ] Add to the changelog_dev.rst file and propose migration text in the pull request

### CODE REVIEW

File: buildstockbatch/localdocker.py [hunk 1]
```diff
@@ -114,7 +114,9 @@ def run_building(cls, project_dir, buildstock_dir, weather_dir, docker_image, re
             if not os.path.exists(dir_to_make):
                 os.makedirs(dir_to_make)
 
-        osw = cls.create_osw(cfg, n_datapoints, sim_id, building_id=i, upgrade_idx=upgrade_idx)
+        output_dir = '../../../run'
+        hpxml_path = '../../../run/in.xml'
+        osw = cls.create_osw(cfg, n_datapoints, sim_id, building_id=i, hpxml_path=hpxml_path, output_dir=output_dir, upgrade_idx=upgrade_idx)
```
Comment (joseph-robertson) [inline buildstockbatch/localdocker.py]: We need to pass in `output_dir` and `hpxml_path` only when using the `residential_hpxml` workflow generator. That, or hardcode `hpxml_path` and `output_dir` in the workflow generator file.

File: buildstockbatch/workflow_generator/residential_hpxml.py [hunk 2]
```diff
@@ -31,14 +31,14 @@ def validate(cls, cfg):
         :type cfg: dict
         """
         schema_yml = """
-        measures_to_ignore: list(str(), required=False)
```
Comment (joseph-robertson) [inline buildstockbatch/workflow_generator/residential_hpxml.py:34]: This is no longer valid for this workflow generator.

File: buildstockbatch/workflow_generator/residential_hpxml.py [hunk 3]
```diff
@@ -31,14 +31,14 @@ def validate(cls, cfg):
...
         build_existing_model: map(required=False)
         emissions: list(include('emission-scenario-spec'), required=False)
         utility_bills: list(include('utility-bill-scenario-spec'), required=False)
         measures: list(include('measure-spec'), required=False)
         reporting_measures: list(include('measure-spec'), required=False)
         simulation_output_report: map(required=False)
         server_directory_cleanup: map(required=False)
+        debug: bool(required=False)
```
Comment (joseph-robertson) [inline buildstockbatch/workflow_generator/residential_hpxml.py:41]: Introducing a "top level" debug argument that controls debug mode for: `HPXMLtoOpenStudio`, `UpgradeCosts`, and `ServerDirectoryCleanup`.

File: buildstockbatch/workflow_generator/residential_hpxml.py [hunk 4]
```diff
@@ -101,17 +101,29 @@ def create_osw(self, sim_id, building_id, upgrade_idx):
...
-        sim_ctl_args = {}
+        sim_ctl_args = {
+            'simulation_control_timestep': 60,
+            'simulation_control_run_period_begin_month': 1,
+            'simulation_control_run_period_begin_day_of_month': 1,
+            'simulation_control_run_period_end_month': 12,
+            'simulation_control_run_period_end_day_of_month': 31,
+            'simulation_control_run_period_calendar_year': 2007,
+            'add_component_loads': False
+        }
```
Comment (joseph-robertson) [inline buildstockbatch/workflow_generator/residential_hpxml.py:112]: Probably should've had these defaults before.

File: buildstockbatch/workflow_generator/residential_hpxml.py [hunk 5]
```diff
@@ -101,17 +101,29 @@ def create_osw(self, sim_id, building_id, upgrade_idx):
...
-        if 'measures_to_ignore' in workflow_args:
-            bld_exist_model_args['measures_to_ignore'] = '|'.join(workflow_args['measures_to_ignore'])
+
         bld_exist_model_args.update(sim_ctl_args)
         bld_exist_model_args.update(workflow_args['build_existing_model'])
 
+        add_component_loads = False
+        if 'add_component_loads' in bld_exist_model_args:
+            add_component_loads = bld_exist_model_args['add_component_loads']
+            bld_exist_model_args.pop('add_component_loads')
```
Comment (joseph-robertson) [inline buildstockbatch/workflow_generator/residential_hpxml.py:125]: This is now handled _outside_ of `BuildExistingModel`.

File: buildstockbatch/workflow_generator/residential_hpxml.py [hunk 6]
```diff
@@ -192,6 +204,10 @@ def create_osw(self, sim_id, building_id, upgrade_idx):
 
         osw['steps'].extend(workflow_args['measures'])
 
+        debug = False
+        if 'debug' in workflow_args:
+            debug = workflow_args['debug']
```
Comment (joseph-robertson) [inline buildstockbatch/workflow_generator/residential_hpxml.py:209]: Again, now handled outside of `BuildExistingModel`.

File: buildstockbatch/workflow_generator/residential_hpxml.py [hunk 7]
```diff
@@ -210,11 +226,21 @@ def create_osw(self, sim_id, building_id, upgrade_idx):
...
         osw['steps'].extend([
+            {
+                'measure_dir_name': 'HPXMLtoOpenStudio',
+                'arguments': {
+                    'hpxml_path': '../../run/home.xml',
+                    'output_dir': '../../run',
+                    'debug': debug,
+                    'add_component_loads': add_component_loads
+                }
+            },
```
Comment (joseph-robertson) [inline buildstockbatch/workflow_generator/residential_hpxml.py:243]: Calling `HPXMLtoOpenStudio` after `BuildExistingModel` and `ApplyUpgrade`. The `hpxml_path` is written by `BuildExistingModel` and (optionally) overwritten by `ApplyUpgrade`.

File: buildstockbatch/workflow_generator/test_workflow_generator.py [hunk 8]
```diff
@@ -359,7 +359,7 @@ def test_residential_hpxml(mocker):
     osw = osw_gen.create_osw(sim_id, building_id, upgrade_idx)
 
     steps = osw['steps']
-    assert len(steps) == 7
+    assert len(steps) == 8
```
Comment (joseph-robertson) [inline buildstockbatch/workflow_generator/test_workflow_generator.py:362]: Our workflow steps has increased by one (i.e., `HPXMLtoOpenStudio`).

File: docs/workflow_generators/residential_hpxml.rst [hunk 9]
```diff
@@ -91,13 +94,18 @@ Arguments
...
   various simulation output files. These arguments are passed directly to
   the `ServerDirectoryCleanup`_ measure in resstock. Please refer to the
   measure arguments there to determine what to set them to in your config file.
   Note that the default behavior is to retain some files and remove others.
   See :ref:`server-dir-cleanup-defaults` for current defaults.
 
+- ``debug`` (optional): Optionally enable debug mode. Enabling debug
+  mode will preserve all simulation input and output files, including but
+  not limited to: in.osm, all EnergyPlus output files, and intermediate
+  existing and upgraded files (e.g., OSWs and XMLs).
```
Comment (joseph-robertson) [inline docs/workflow_generators/residential_hpxml.rst:107]: Documenting the new top level debug argument that can optionally be called out in the yml file.

**Review by aspeake — APPROVED** [review]

**Pull request merged**